packages/telemetry/src/react/index.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseOptions, initializeApp } from '@firebase/app';
+import { registerTelemetry } from '../register';
+import { captureError, getTelemetry } from '../api';
+import { Component, ReactNode } from 'react';
+
+registerTelemetry();
+
+export interface FirebaseTelemetryProps {
+  firebaseOptions?: FirebaseOptions;
+  children?: ReactNode;
+}
+
+export class FirebaseTelemetry extends Component<FirebaseTelemetryProps> {
```

Comment:
Isn't the class-based component style with lifecycle methods not recommended for React anymore? https://react.dev/reference/react/Component